lib/puppet/provider/package/pip.rb

```diff
@@ -29,7 +29,11 @@ def self.parse(line)
   # that's managed by `pip` or an empty array if `pip` is not available.
   def self.instances
     packages = []
-    pip_cmd = which(cmd) or return []
+    if cmd.class == Array
+      pip_cmd = ( which(cmd.first) or which(cmd.last) ) or return []
```

Comment:
This makes an assumption that there are always two elements in the array.  A cleaner implementation would be:

```
...
pip_cmd = cmd.find { |c| which(c) }
return [] unless pip_cmd
...
``

We prefer not to use the `and` and `or` keywords as they can introduce unintentional behavior when used in logical expressions where developers don't expect their precedence compared to `&&` and `||`.  Additionally, using them for control flow, such as returning early from a function, makes it a little harder to spot what's happening.